knotx-core/src/main/java/io/knotx/options/SnippetOptions.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2016 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.options;
+
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Describes SnippetOptions Knot configuration
+ */
+@DataObject(generateConverter = true, publicConverter = false)
+public class SnippetOptions {
+
+  private String tagName;
+  private String paramsPrefix;
+
+  /**
+   * Default constructor
+   */
+  public SnippetOptions() {
+  }
```

Comment:
what about default values ? a good practice is to have private init() method invoked in default constructor and inside jsonobject constructor (before converter). an init method will simply set default values for tag and prefix. that way we avoid potential nullpointers